### PR TITLE
Fixes #5673: Add support to parameters in ncf techniques written with ncf builder

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/5673

### CODE REVIEW

File: builder/js/ncf.js [hunk 1]
```diff
@@ -994,6 +995,15 @@ $scope.groupMethodsByCategory = function () {
     $scope.saving = false;
   });
 
+  $scope.newParam = {
+    "description" : "test"
+  }
+  $scope.addParameter = function() {
+    console.log($scope.newParam)
```
Comment (peckpeck) [inline builder/js/ncf.js]: Debug info ?

**Comment by peckpeck** [comment]
LGTM

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

File: builder/js/ncf.js [hunk 2]
```diff
@@ -994,6 +995,12 @@ $scope.groupMethodsByCategory = function () {
     $scope.saving = false;
   });
 
+  $scope.newParam = {}
+  
+  $scope.addParameter = function() {
+    $scope.selectedTechnique.parameter.push(angular.copy($scope.newParam))
```
Comment (peckpeck) [inline builder/js/ncf.js]: I don't understand how and when newParam is modified

File: tools/ncf.py [hunk 3]
```diff
@@ -513,9 +522,20 @@ def generate_technique_content(technique, methods):
     # Add commentary for each new line in the metadata
     metadata_value = technique[metadata_key].replace("\n", "\n# ")
     content.append('# @'+ metadata_key +" "+ metadata_value)
+
+  bundle_param = ""
+  if technique["parameter"] > 0:
```
Comment (peckpeck) [inline tools/ncf.py]: Why do you compare a list with an integer

File: tools/ncf.py [hunk 4]
```diff
@@ -513,9 +522,20 @@ def generate_technique_content(technique, methods):
...
+    bundle_param = "("+", ".join([re.sub("\W","_",param["name"]) for param in technique["parameter"] ])+")"
```
Comment (peckpeck) [inline tools/ncf.py]: Please use r"" or r'' insdeat of "" when writing a litteral regex.

Comment (peckpeck) [reply tools/ncf.py]: Be careful with \W
It doesn't have the same behaviour depending on the locale and on the ancoding on the string.
And the default encoding of the string is not the same in python 2 and python 3

I think we already have a canoncalization function in python somewhere

File: builder/js/ncf.js (hunk 2 above)
Comment (VinceMacBuche) [reply builder/js/ncf.js]: In fact newParam does not need to exists anymore and could be replaced by an empty object to simplify things here

File: tools/ncf.py (hunk 3 above)
Comment (VinceMacBuche) [reply tools/ncf.py]: forgot a len ...

**Comment by VinceMacBuche** [comment]
Commit modified

**Pull request merged**